apps/designer/app/designer/shared/design-tokens-manager/design-tokens-manager.tsx

```diff
@@ -79,108 +104,230 @@ const TokenEditor = ({
+                  defaultValue={"name" in token ? token.name : ""}
+                />
               </InputErrorsTooltip>
               <Label htmlFor="value">Value</Label>
               <InputErrorsTooltip
                 errors={errors.value}
                 css={{ zIndex: "$zIndices$2" }}
               >
-                <TextField id="value" name="value" />
+                <TextField
+                  id="value"
+                  name="value"
+                  defaultValue={"value" in token ? token.value : ""}
+                />
               </InputErrorsTooltip>
               <Label htmlFor="description">Description</Label>
-              <TextArea id="description" name="description" />
+              <TextArea
+                id="description"
+                name="description"
+                defaultValue={"description" in token ? token.description : ""}
+              />
               {isNew && (
                 <Button type="submit" variant="blue">
                   Create
                 </Button>
               )}
             </Flex>
           </form>
-          <PopoverHeader title={isNew ? "New Token" : name} />
+          <PopoverHeader title={"name" in token ? token.name : "New Token"} />
         </PopoverContent>
       </PopoverPortal>
     </Popover>
   );
 };
 
-const TokenItem = ({ token }: { token: DesignToken }) => {
-  return <div>{token.value}</div>;
-};
-
 export const DesignTokensManager = ({ publish }: { publish: Publish }) => {
   const [tokens, setTokens] = useDesignTokens();
+  const [selectedIndex, setSelectedIndex] = useState(-1);
+  const [currentIndex, setCurrentIndex] = useState(-1);
+  const { getItemProps, getListProps } = useList({
+    items: tokens,
+    selectedIndex,
+    currentIndex,
+    onSelect: setSelectedIndex,
+    onChangeCurrent: setCurrentIndex,
+  });
+  const [editingToken, setEditingToken] = useState<
+    Partial<DesignToken> | undefined
+  >();
+  const { render: renderMenu, isOpen: isMenuOpen } = useMenu({
+    selectedIndex,
+    onSelect: setSelectedIndex,
+    onDelete: () => {
+      const { name } = tokens[selectedIndex];
+      publish({ type: "deleteToken", payload: name });
+      const updatedTokens = [...tokens];
+      deleteTokenMutable(updatedTokens, name);
+      setTokens(updatedTokens);
+    },
+    onEdit: (index) => {
+      setEditingToken(tokens[index]);
+    },
+  });
+  const listProps = getListProps();
+
+  const renderTokenEditor = ({
+    token,
+    isOpen,
+    trigger,
+  }: {
+    token: DesignToken | DesignTokenSeed;
+    isOpen: boolean;
+    trigger?: JSX.Element;
+  }) => {
+    return (
+      <TokenEditor
+        token={token}
+        isOpen={isOpen}
+        trigger={trigger}
+        onChangeComplete={(token) => {
+          const foundToken = findByName(tokens, token.name);
+          const previousTokenName = token.name;
+          const isNew =
+            foundToken === undefined || previousTokenName === undefined;
```

Comment:
I am actually advocating to always use undefined and remove the need for null checks, by using == you imply supporting both null and undefined, but that's not the case here and neither would I want any of our apis to expose interfaces with null, but rather actually try to contain them in the local usages where we forced to use null with dom/react apis

Reply:
reason name is not set is because its the `token` it references is TokenSeed, basically token is {group, type, name, value, descript}, seed is {group, type} ... group and type are predefined, the rest is user defined.

would you structure it differently?

Reply:
> Why can't we force for the token to always have a defined name?

We could have make the Token to always have all the fields with empty strings, but token name has to be unique and can't be empty, so it would not match DesignToken schema if I run it over it

Reply:
separated seed from token so that there is no semantic confusion